File: 1615-maximal-network-rank/1615-maximal-network-rank.java
class Solution {
    public int maximalNetworkRank(int n, int[][] roads) {
        Map<Integer,Set<Integer>> map = new HashMap();
        for (int i=0;i<n;i++)
            map.put(i,new HashSet());
        for (int[] r : roads) {
            map.get(r[0]).add(r[1]);
            map.get(r[1]).add(r[0]);
        }
        
        int max = 0;
        for (int i=0;i<n;i++)
            for (int j=i+1;j<n;j++)
            {
                int a = map.get(i).size();
                int b = map.get(j).size();
                int cur = a+b;
                //System.out.println(i+" "+a+" "+j+" "+b);
                if ( map.get(j).contains(i) ) cur--;
                max = Math.max(cur,max);
            }
                
        return max;
        
    }
}

File: 1615-maximal-network-rank/1615-maximal-network-rank.py
class Solution:
    def maximalNetworkRank(self, n: int, roads: List[List[int]]) -> int:
        map = {}
        for i in range(n):
            map[i] = []
        for r in roads:
            map[r[0]].append(r[1])
            map[r[1]].append(r[0])
        
        
        
        c = 0
        
        for i in range(n):
            for j in range(i+1,n):
                a = len(map[i])
                b = len(map[j])
                cur = a + b
                if i in map[j]:
                    cur-=1
                c = max(c,cur)
        
        return c